ibm/service/cos/resource_ibm_cos_bucket_objectlock_configuration_test.go

```diff
@@ -0,0 +1,755 @@
+	}
+	resource "ibm_cos_bucket" "bucket" {
+		bucket_name           = "%s"
+		resource_instance_id  = ibm_resource_instance.instance.id
+	    cross_region_location = "%s"
+		storage_class         = "%s"
+		object_versioning {
+			enable  = true
+		}
+		object_lock = "%t"
+	}
+	resource ibm_cos_bucket_objectlock_configuration "objectlock" {
+		bucket_crn      = ibm_cos_bucket.bucket.crn
+		bucket_location = ibm_cos_bucket.bucket.cross_region_location
+		object_lock_configuration{
+			objectlockenabled = "Enabled"
+			objectlockrule{
+			  defaultretention{
+				mode = "%s"
+				days = "%d"
+				}
+			}
+		  }   
+	   }
+	`, cosServiceName, bucketName, region, storageClass, objectLockEnabled, mode, days)
+}
+
+func testAccCheckIBMCosBucket_Objectlock_Valid_Mode_and_Years(cosServiceName string, bucketName string, regiontype string, region string, storageClass string, objectLockEnabled bool, mode string, years int64) string {
+
+	return fmt.Sprintf(`
+	data "ibm_resource_group" "cos_group" {
+		name = "Default"
+	}
+
+	resource "ibm_resource_instance" "instance" {
+		name              = "%s"
+		service           = "cloud-object-storage"
+		plan              = "standard"
+		location          = "global"
+		resource_group_id = data.ibm_resource_group.cos_group.id
+	}
+	resource "ibm_cos_bucket" "bucket" {
+		bucket_name           = "%s"
+		resource_instance_id  = ibm_resource_instance.instance.id
+	    cross_region_location = "%s"
+		storage_class         = "%s"
+		object_versioning {
+			enable  = true
+		}
+		object_lock = "%t"
+	}
+	resource ibm_cos_bucket_objectlock_configuration "objectlock" {
+		bucket_crn      = ibm_cos_bucket.bucket.crn
+		bucket_location = ibm_cos_bucket.bucket.cross_region_location
+		object_lock_configuration{
+			objectlockenabled = "Enabled"
+			objectlockrule{
+			  defaultretention{
+				mode = "%s"
+				years = "%d"
+				}
+			}
+		  }   
+	   }
+	`, cosServiceName, bucketName, region, storageClass, objectLockEnabled, mode, years)
+}
+
+func testAccCheckIBMCosBucket_Objectlock_Empty(cosServiceName string, bucketName string, regiontype string, region string, storageClass string, objectLockEnabled bool) string {
+
+	return fmt.Sprintf(`
+	data "ibm_resource_group" "cos_group" {
+		name = "Default"
+	}
+
+	resource "ibm_resource_instance" "instance" {
+		name              = "%s"
+		service           = "cloud-object-storage"
+		plan              = "standard"
+		location          = "global"
+		resource_group_id = data.ibm_resource_group.cos_group.id
+	}
+	resource "ibm_cos_bucket" "bucket" {
+		bucket_name           = "%s"
+		resource_instance_id  = ibm_resource_instance.instance.id
+	    cross_region_location = "%s"
+		storage_class         = "%s"
+		object_versioning {
+			enable  = true
+		}
+		object_lock = "%t"
+	}
+	resource ibm_cos_bucket_objectlock_configuration "objectlock" {
+		bucket_crn      = ibm_cos_bucket.bucket.crn
+		bucket_location = ibm_cos_bucket.bucket.cross_region_location
+		object_lock_configuration{
```

Comment:
why is the structure empty?

Reply:
This is a negative test case to test a scenario where the configuration is  empty.